internal/cmd/egctl/translate.go

```diff
@@ -632,14 +632,25 @@ func addMissingServices(requiredServices map[string]*corev1.Service, obj interfa
 	}
 }
 
+// LoadResources struct to handle multiple GCs and EPs
+type LoadResources struct {
```

Comment:
The structure looks good! But we cannot directly use it, we may need to create N `gatewayapi.Resources` according to how many GCs you have collected, and pass each generated `gatewayapi.Resources` to each translate function.